File: front-end/src/app/core/services/dtos/shipping-address-dto.ts
/** Shipping address DTO. */
export interface ShippingAddressDto {
  /** Uniquely identify a shipping address. */
  readonly id: number | null;

  /** User full name. */
  readonly fullname: string;

  /** User country. */
  readonly country: string;

  /** User phone number. */
  readonly phoneNumber: string;

  /** User address. */
  readonly address: string;

  /** User city. */
  readonly city: string;

  /** User note. */
  readonly note: string | null;
}

File: back-end/src/main/java/com/huyphan/dtos/ShippingAddressDto.java
package com.huyphan.dtos;

import com.huyphan.models.enums.Country;
import lombok.Getter;
import lombok.Setter;

/** Shipping address DTO. */
@Getter
@Setter
public class ShippingAddressDto {

	/** User full name. */
	private String fullname;

	/** User country. */
	private Country country;

	/** User phone number. */
	private String phoneNumber;

	/** User address. */
	private String address;

	/** User city. */
	private String city;

	/** User note. */
	private String note;
}
